### PR TITLE
Build: correct emitted output file map

### PR DESCRIPTION
**Pull request by compnerd**
This adjusts the output file map writing code to ensure that we emit the
proper native path string and correct escaping of `\` which otherwise
yields invalid JSON which cannot be deserialized subsequently.  This
will be critical to enabling the use of swift-driver with SPM.

_[One line description of your change]_

### Motivation:

_[Explain here the context, and why you're making that change. What is the problem you're trying to solve.]_

### Modifications:

_[Describe the modifications you've done.]_

### Result:

_[After your change, what will change.]_

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: Sources/Build/BuildPlan.swift [hunk 1]
```diff
@@ -25,6 +25,18 @@ extension String {
     }
 }
 
+extension String {
```
Comment (tomerd) [inline Sources/Build/BuildPlan.swift:28]: fells like this should be extensions on AbsolutePath rather than on String

Comment (compnerd) [reply Sources/Build/BuildPlan.swift:28]: As a local extension to `AbsolutePath` is what you have in mind?

Comment (tomerd) [reply Sources/Build/BuildPlan.swift:28]: right, this basically operates on the string version of absolute path, so the call sites are `<path>.pathString.nativePathString()` and would be better as  `<path>.nativePathString()`

Comment (tomerd) [reply Sources/Build/BuildPlan.swift:28]:
```
extension AbsolutePath {
  internal func nativePathString() -> String {
       return URL(fileURLWithPath: self.pathString).withUnsafeFileSystemRepresentation {
           String(cString: $0!)
       }
   }
}
```

Comment (compnerd) [reply Sources/Build/BuildPlan.swift:28]: Agreed, just wanted to make sure that the location was still within this file :)

Comment (tomerd) [reply Sources/Build/BuildPlan.swift:28]: also consider computed var vs. func to make it more like `pathString`, tho I can go both ways given the allocations involved

Comment (tomerd) [reply Sources/Build/BuildPlan.swift:28]: one more suggestion: make it into `nativePathString(escaped: Bool)` so this way we can collapse these two into one clean API

**Review by tomerd — APPROVED** [review]

**Review by neonichu — APPROVED** [review]

**Pull request merged**